src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/FrameOfT.cs

```diff
@@ -40,26 +47,47 @@ public Frame(IHttpApplication<TContext> application, FrameContext frameContext)
 
                     while (!_requestProcessingStopping)
                     {
-                        var result = await Input.ReadAsync();
-                        var examined = result.Buffer.End;
-                        var consumed = result.Buffer.End;
+                        if (needBuffer)
+                        {
+                            result = await Input.ReadAsync();
+                            buffer = result.Buffer;
+                            needBuffer = false;
+                        }
+
+                        var needAdvance = false;
 
                         try
                         {
-                            ParseRequest(result.Buffer, out consumed, out examined);
+                            ParseRequest(ref buffer, out consumed, out examined);
+
+                            if (buffer.Length == 0)
```

Comment:
My bad. We had it until I converted a bunch of tests to parser tests: https://github.com/aspnet/KestrelHttpServer/commit/5743d740b454874a3d056d2f480297be51abcb71#diff-78adb33626b9b8aa43b7d4529f4f0bec

I'm bringing it back.